### ISSUE TITLE
RDG should use render funcs instead of components

### ISSUE THREAD

**Issue by alamothe** [adazzle/react-data-grid#2872]
Currently, RDG relies on components to customize rendering:
```tsx
  /** Formatter to be used to render the cell content */
  readonly formatter?: Maybe<ComponentType<FormatterProps<TRow, TSummaryRow>>>;
  /** Formatter to be used to render the summary cell content */
  readonly summaryFormatter?: Maybe<ComponentType<SummaryFormatterProps<TSummaryRow, TRow>>>;
  /** Formatter to be used to render the group cell content */
  readonly groupFormatter?: Maybe<ComponentType<GroupFormatterProps<TRow, TSummaryRow>>>;
```

This is problematic:
- Customization is more difficult, we have to create a component every time, but sometimes can be easily solved with a render func
- Error prone, as a lambda can be passed. This prevents React from reusing the DOM and causes a performance penalty.

Instead, RDG should require and use render funcs:
```tsx
  /** Formatter to be used to render the cell content */
  readonly formatter?: Maybe<(props: FormatterProps<TRow, TSummaryRow>) => React.ReactNode>;
  /** Formatter to be used to render the summary cell content */
  readonly summaryFormatter?: Maybe<(props: SummaryFormatterProps<TSummaryRow, TRow>) => React.ReactNode>;
  /** Formatter to be used to render the group cell content */
  readonly groupFormatter?: Maybe<(props: GroupFormatterProps<TRow, TSummaryRow>) => React.ReactNode>;
```
Note that with this approach, should one want, they can still use a component easily:
```tsx
formatter: (props) => <MyComponent {...props} />
```

React libraries have been moving to this approach. In particular, `react-router` has switched from 4 => 5. Please read more about it here: https://reactrouter.com/docs/en/v6/upgrading/v5#advantages-of-route-element

**Comment by nstepien** [comment]
I agree, using components is leading to some performance issues for us internally, and requires us to be more thoughtful than I'd like.
I'll try to do the change.
CC @amanmahajan7

**Comment by alamothe** [comment]
This change would be amazing! hope it will go out in the next beta. Thanks for the awesome library

**Comment by nstepien** [comment]
I've published `v7.0.0-beta.15` with the change 👍
